typescript/taxonomy-api.ts

```diff
@@ -0,0 +1,905 @@
+    /**
+     * The name of the relevance
+     */
+    name: string;
+    /**
+     * All translations of this relevance
+     */
+    translations: Translation[];
+    /**
+     * List of language codes supported by translations
+     */
+    supportedLanguages: string[];
+}
+
+export interface RelevancePUT {
+    /**
+     * If specified, set the id to this value. Must start with urn:relevance: and be a valid URI. If ommitted, an id will be assigned automatically. Ignored on update
+     */
+    id: string;
+    /**
+     * The name of the relevance
+     */
+    name: string;
+}
+
+export interface ResourceResourceTypes {
+    allResourceResourceTypes: ResourceResourceType[];
+}
+
+export interface ResourceResourceType {
+    /**
+     * Resource type id
+     */
+    resourceId: string;
+    /**
+     * Resource type id
+     */
+    resourceTypeId: string;
+    /**
+     * Resource to resource type connection id
+     */
+    id: string;
+}
+
+export interface ResourceResourceTypePOST {
+    /**
+     * Resource id
+     */
+    resourceId: string;
+    /**
+     * Resource type id
+     */
+    resourceTypeId: string;
+}
+
+/**
+ * @deprecated
+ */
+export interface ResourceTranslations {
+}
+
+export interface ResourceTranslation {
+    /**
+     * The translated name of the resource
+     */
+    name: string;
+    /**
+     * ISO 639-1 language code
+     */
+    language: string;
+}
+
+export interface ResourceTranslationPUT {
+    /**
+     * The translated name of the resource
+     */
+    name: string;
+}
+
+export interface ResourceTypeTranslations {
+}
+
+export interface ResourceTypeTranslation {
+    /**
+     * The translated name of the resource type
+     */
+    name: string;
+    /**
+     * ISO 639-1 language code
+     */
+    language: string;
+}
+
+export interface ResourceTypeTranslationPUT {
+    /**
+     * The translated name of the resource type
+     */
+    name: string;
+}
+
+export interface ResourceTypes extends CrudController<any> {
+}
+
+export interface ResourceType {
+    id: string;
+    /**
+     * The name of the resource type
+     */
+    name: string;
+    /**
+     * Sub resource types
+     */
+    subtypes: ResourceType[];
+    /**
+     * All translations of this resource type
+     */
+    translations: Translation[];
+    /**
+     * List of language codes supported by translations
+     */
+    supportedLanguages: string[];
+}
+
+export interface ResourceTypePUT {
+    /**
+     * If specified, the new resource type will be a child of the mentioned resource type.
+     */
+    parentId: string;
+    /**
+     * If specified, set the id to this value. Must start with urn:resourcetype: and be a valid URI. If omitted, an id will be assigned automatically.
+     */
+    id: string;
+    /**
+     * The name of the resource type
+     */
+    name: string;
+}
+
+/**
+ * @deprecated
+ */
+export interface Resources extends CrudControllerWithMetadata<any> {
+}
+
+/**
+ * @deprecated
+ */
```

Comment:
Stemmer det at `@deprecated` legges på roten av controlleren? Isåfall vil dette bli feil i TS.

Reply:
Har lagt på rota. Kan heller legge på metodenivå om det er bedre.

Reply:
Flytta til metodenivå. Har forresten ikkje lenger betydning for typescipt då controllere ikkje lenger er med i fila.